lib/concurrent/utility/monotonic_time.rb

```diff
@@ -0,0 +1,147 @@
+module Concurrent
+
+  # Clock that cannot be set and represents monotonic time since
+  # some unspecified starting point.
+  # @!visibility private
+  GLOBAL_MONOTONIC_CLOCK = Class.new {
+
+    if defined?(Process::CLOCK_MONOTONIC)
+      # @!visibility private
+      def get_time
+        Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      end
+    elsif RUBY_PLATFORM == 'java'
+      # @!visibility private
+      def get_time
+        java.lang.System.nanoTime() / 1_000_000_000.0
+      end
+    else
+
+      require 'thread'
+
+      # @!visibility private
+      def initialize
+        @mutex = Mutex.new
+        @correction = 0
+        @last_time = Time.now.to_f
+      end
+
+      # @!visibility private
+      def get_time
+        @mutex.synchronize {
+          @correction ||= 0 # compensating any back time shifts
+          now = Time.now.to_f
+          corrected_now = now + @correction
+          if @last_time < corrected_now
+            @last_time = corrected_now 
+          else
+            @correction += @last_time - corrected_now + 0.000_001
+            @last_time = @correction + now
+          end
+        }
+      end
+    end
+  }.new
```

Comment:
The reason I used `{}` for that block is that I really dislike calling methods on an `end` statement. The natural replacement for that line would be `end.new` and that syntax always seems difficult to read. Since the convention of using `begin/end` for multiline blocks is just that--a convention--and not universally accepted by all Rubyists, I went with the syntax that made the most sense to be in this case.

If the team wants to develop a style guide for this gem (or adopt one created by another project) I'm happy to support that. I've never rejected a PR based on style, nor have I ever rejected a PR because someone changed the style of existing code. It's just something that's never been a high priority for me.